router/query.go

```diff
@@ -52,15 +54,23 @@ func QueryRouter(r string) QueryOption {
 	}
 }
 
+// QueryStrategy sets strategy to query
+func QueryStrategy(s Strategy) QueryOption {
+	return func(o *QueryOptions) {
+		o.Strategy = s
+	}
+}
+
 // NewQuery creates new query and returns it
 func NewQuery(opts ...QueryOption) QueryOptions {
 	// default options
 	qopts := QueryOptions{
-		Service: "*",
-		Address: "*",
-		Gateway: "*",
-		Network: "*",
-		Router:  "*",
+		Service:  "*",
+		Address:  "*",
+		Gateway:  "*",
+		Network:  "*",
+		Router:   "*",
+		Strategy: AdvertiseAll,
```

Comment:
This should probably default to whatever the routes own strategy is so that in the event you do not specify you actually get back default e.g core nodes advertise best and a query returns best, edge nodes advertise local and this returns local.

Reply:
But how do we know that at this point? Router's strategy is set in Router's options. 

This function picks default values before it applies the value the user passes in.

Reply:
Guess you're right